test/test_parse_eu.py

```diff
@@ -0,0 +1,231 @@
+#
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from datetime import datetime
+import unittest
+
+from lingua_franca.parse import (normalize, extract_numbers, extract_number,
+                                 extract_datetime, extract_datetime_eu,
+                                 isFractional_eu)
+
+
+class TestNormalize(unittest.TestCase):
+    """
+        Test cases for Euskara parsing
+    """
+    # TODO: Hau ez dakit behar dugun
+    def test_articles_eu(self):
+        self.assertEqual(normalize("hau da froga", lang="eu",
+                                   remove_articles=True),
+                         "hau da froga")
+        self.assertEqual(normalize("eta hau beste froga", lang="eu",
+                                   remove_articles=True),
+                         "eta hau beste froga")
+
+    def test_numbers_eu(self):
+        self.assertEqual(normalize("hau da bat", lang="eu"),
+                         "hau da 1")
+        self.assertEqual(normalize("hau da bi hiru froga", lang="eu"),
+                         "hau da 2 3 froga")
+        self.assertEqual(normalize("hau da lau bost sei froga",
+                                   lang="eu"),
+                         "hau da 4 5 6 froga")
+        self.assertEqual(normalize("zazpi mï¿½s zortzi mï¿½s bederatzi", lang="eu"),
+                         "7 mï¿½s 8 mï¿½s 9")
+        self.assertEqual(normalize("hamar hamaika hamabi hamahiru hamalau hamabost",
+                                   lang="eu"),
+                         "10 11 12 13 14 15")
+        self.assertEqual(normalize("hamasei hamazazpi", lang="eu"),
+                         "16 17")
+        self.assertEqual(normalize("hemezortzi hemeretzi", lang="eu"),
+                         "18 19")
+        self.assertEqual(normalize("hogei hogeita hamar berrogeita bat", lang="eu"),
+                         "20 30 41")
+        self.assertEqual(normalize("hogeita hamabi zaldi", lang="eu"),
+                         "32 zaldi")
+        self.assertEqual(normalize("ehun zaldi", lang="eu"),
+                         "100 zaldi")
+        self.assertEqual(normalize("ehun eta hamaika zaldi", lang="eu"),
+                         "111 zaldi")
+        self.assertEqual(normalize("laurehun eta bat behi zeuï¿½en",
+                                   lang="eu"),
+                         "401 behi zeuï¿½en")
```

Comment:
I see a few cases with what appear to be the wrong unicode characters. Is this a font thing?

Reply:
There are occurrences of that character in the portuguese and spanish files, and I think the reason for their being in the basque files is because they were taken as a base. I have removed and corrected them from the basque files and also from the others, I am pretty confident they were there due to wrong encoding from copy/pasting and not for correcting input in another encoding (it's just in a couple of cases from many possible ones).

Reply:
Welcome to the glorious world of special characters in code ;) kicked back up near the top of my code review list, thanks for addressing it quickly